delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphGlobalEventLog.scala

```diff
@@ -56,7 +55,18 @@ final class BlazegraphGlobalEventLog private (
       .map(_.toMessage)
       .groupWithin(batchMaxSize, batchMaxTimeout)
       .discardDuplicates()
-      .evalMapFilterValue(event => eventExchanges.findFor(event).traverse(_.toState(event, tag).flatMap(_.toGraph)))
+      .evalMapFilterValue { event =>
+        Stream
+          .fromIterator[Task](referenceExchanges.iterator)
+          .evalMap { _(event, tag) }
```

Comment:
`evalMapFilter` == evalMap + collect

Reply:
This is far less efficient than what we had with `EventExchangeCollection`, since for every event you are blindly going to attempt each element of the list of modules we have.

The `EventExchangeCollection` had a cache where after a while all event classes would be present and then we would have direct access to its exact `EventExchange` (in this case `ReferenceExchange`).

I feel that we need the same logic here. I understand we don't need to have it right now in this PR, but should be done after. If you don't want to do it now, please add at least a TODO